rules/coding-style/src/Rector/String_/UseClassKeywordForClassNameResolutionRector.php

```diff
@@ -50,23 +50,40 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        if (substr_count($node->value, '::') !== 1) {
-            return null;
-        }
-        // a possible static call reference
         // @see https://regex101.com/r/Vv41Qr/1/
-        [$before, $possibleClass, $after] = Strings::split($node->value, '#([\\\\a-zA-Z0-9_\\x80-\\xff]*)::#');
+        $matches = Strings::matchAll($node->value, '#([\\\\a-zA-Z0-9_\\x80-\\xff]*)::#', PREG_PATTERN_ORDER);
+
+        $classNames = array_filter($matches[1], function (string $className) {
+            return class_exists($className);
+        });
 
-        if (! $possibleClass || ! class_exists($possibleClass)) {
-            return null;
+        if (empty($classNames)) {
+            return $node;
         }
 
-        $classConstFetch = new ClassConstFetch(new FullyQualified(ltrim($possibleClass, '\\')), 'class');
+        $classNames = array_map(function (string $className) {
+            return preg_quote($className);
+        }, $classNames);
+
+        // @see https://regex101.com/r/8nGS0F/1
+        $splits = Strings::split($node->value, '#(' . implode('|', $classNames) . ')#');
+
+        foreach ($splits as $split) {
+            if (empty($split)) {
+                continue;
+            }
+
+            if (class_exists($split)) {
+                $returnNode = new ClassConstFetch(new FullyQualified(ltrim($split, '\\')), 'class');
+            } else {
+                $returnNode = new String_($split);
+            }
+
+            $concat = ! isset($concat) ? $returnNode : new Concat($concat, $returnNode);
```

Comment:
This looks very crazy :smile:  What is going on here? 
It's some node stacking in a loop?

Better collect concat nodes in an array, e.g. to $contacts

```php
$concats[] = $concat;
```

And resolve later in in go, e.g.

```php
if ($concats === []) {
    return ...;
}

return ...;
```


It's much easier to debug and test in case of troubles.

Reply:
This solution helped me to keep myself away from becoming crazy :D Believe me or not but is a million percents better than a couple of the solution I had before this one :D

Maybe it is not clear but this solution is able to process the random amount of class name occurrences in the string (not only one or two) that is why concatenation in the loop is required.

When it comes to the condition it helps to avoid the first empty Node. If I won't use it I'll have to create the first Node with an empty value outside of the loop for first concatenation which will then produce empty node which will in final result look like 

```
'' . \App\ClassName()::class . 'staticCall()'
```

While it is not a problem to create the array of nodes with the current solution... I can't imagine what I should implement under `...` in your code to cover the cases in the test fixtures.

Reply:
I see, I think what you mean. Nodes can break human brain sometimes, I've been there :D. 
Don't worry about it :+1: , I'll check if there are is some space after we merge this.

Reply:
Not sure how much you care about perfomance but to make this loop a bit nicer I can also 
```diff
+            $returnNode = new String_($part);
+            if (class_exists($part)) {
+                $returnNode = new ClassConstFetch(new FullyQualified(ltrim($part, '\\')), 'class');
+            }
-            if (class_exists($part)) {
-                $returnNode = new ClassConstFetch(new FullyQualified(ltrim($part, '\\')), 'class');
-            } else {
-                $returnNode = new String_($part);
-            }
```

Reply:
Unless a memory/performance leak, it's not an issue as this is CLI app, not a response/request one.

Overriding variable like this is confusing. Personally I'd extract it to own method too. E.g. `createReturnNode()` with early returns.

But current state is also ok

Reply:
> I see, I think what you mean. Nodes can break human brain sometimes, I've been there :D.
> Don't worry about it  , I'll check if there are is some space after we merge this.

Not sure what do you mean by that. Can I leave it as it looks like at the moment (after today's changes)?